xla/service/cpu/onednn_convolution.cc

```diff
@@ -70,8 +77,140 @@ GetKernelConfig<kOnednnConvConfig>(
   return (*backend_config)->mutable_onednn_conv_config();
 }
 
+template <>
+std::unique_ptr<dnnl::convolution_forward::primitive_desc>
+CreateOneDnnPrimDesc<dnnl::convolution_forward::primitive_desc>(
+    HloInstruction* instr) {
+  if (instr->opcode() != HloOpcode::kCustomCall) {
+    return nullptr;
+  }
+  xla::HloCustomCallInstruction* custom_call =
+      Cast<xla::HloCustomCallInstruction>(instr);
+  absl::StatusOr<BackendConfig> backend_config =
+      custom_call->backend_config<BackendConfig>();
+  if (!backend_config.ok()) {
+    return nullptr;
+  }
+  auto& conv_config = backend_config.value().onednn_conv_config();
+  absl::InlinedVector<HloInstruction*, 2> operands = custom_call->operands();
+  Shape input_shape = operands[0]->shape();
+  Shape weight_shape =
+      operands[1]->shape();  // assuming weights is the second operand
+  Shape output_shape = custom_call->shape().IsTuple()
+                           ? custom_call->shape().tuple_shapes(0)
+                           : custom_call->shape();
+
+  absl::InlinedVector<HloInstruction*, 2> fused_operands =
+      HloInstruction::InstructionVector(operands.begin() + 2, operands.end());
+  std::vector<Shape> fused_shapes;
+  std::transform(fused_operands.begin(), fused_operands.end(),
+                 std::back_inserter(fused_shapes),
+                 [](const HloInstruction* instr) { return instr->shape(); });
```

Comment:
Using a regular for loop is much simpler here IMO:

```suggestion
std::vector<Shape> fused_shapes;
for (int i = 2; i < operands.size(); ++i) {
  fused_shapes.push_back(operands[i]->shape());
}
```

Reply:
Done